parsl/executors/high_throughput/interchange.py

```diff
@@ -672,13 +672,10 @@ def start_file_logger(filename: str, level: int = logging.DEBUG, format_string:
     logger.addHandler(handler)
 
 
-@wrap_with_logs(target="interchange")
-def starter(*args: Any, **kwargs: Any) -> None:
-    """Start the interchange process
-
-    The executor is expected to call this function. The args, kwargs match that of the Interchange.__init__
-    """
+if __name__ == "__main__":
     setproctitle("parsl: HTEX interchange")
-    # logger = multiprocessing.get_logger()
-    ic = Interchange(*args, **kwargs)
+
+    config = pickle.load(sys.stdin.buffer)
+
+    ic = Interchange(**config)
```

Comment:
Is there any reason to keep stdin open after this?  We might consider clearing that fd as soon as we're reasonably done with it.

Reply:
Feels kinda weird closing stdin at all, although should be legit and lets us open another half-a-worker-pool worth of connections (with fds being the limit right now on maximum worker pool connections, and each one requiring two fds), I guess.

Less weird to me is closing at the other end of the pipe in executor.py

Reply:
related, see #3501 which closes the executor end of this pipe